File: jvm/core/src/main/java/se/idkollen/client/IdkollenClient.java
package se.idkollen.client;

import okhttp3.OkHttpClient;
import se.idkollen.client.endpoints.*;
import se.idkollen.client.internal.Transport;

/**
 * Authenticated HTTP client for the IDkollen REST API.
 *
 * <p>Obtain an instance via {@link IdkollenClientBuilder}. All network calls return
 * {@link java.util.concurrent.CompletableFuture} and are non-blocking.
 */
public final class IdkollenClient {
    private final Transport transport;

    IdkollenClient(String baseUrl, String clientId, String clientSecret, String userAgent, OkHttpClient http) {
        this.transport = new Transport(baseUrl, clientId, clientSecret, userAgent, http);
    }

    /** Return the BankID SE endpoint accessor. */
    public BankIdSeEndpoint bankIdSe() {
        return new BankIdSeEndpoint(transport);
    }

    /** Return the BankID NO endpoint accessor. */
    public BankIdNoEndpoint bankIdNo() {
        return new BankIdNoEndpoint(transport);
    }

    /** Return the Freja eID endpoint accessor. */
    public FrejaEndpoint freja() {
        return new FrejaEndpoint(transport);
    }

    /** Return the MitID endpoint accessor. */
    public MitIdEndpoint mitId() {
        return new MitIdEndpoint(transport);
    }

    /** Return the Finnish Trust Network (FTN) endpoint accessor. */
    public FtnEndpoint ftn() {
        return new FtnEndpoint(transport);
    }

    /** Return the Vipps MobilePay endpoint accessor. */
    public VippsEndpoint vipps() {
        return new VippsEndpoint(transport);
    }

    /** Return the document upload/download endpoint accessor. */
    public DocumentEndpoint document() {
        return new DocumentEndpoint(transport);
    }
}

File: csharp/src/Idkollen.Client/IdkollenClient.cs
using Idkollen.Client.Endpoints;

namespace Idkollen.Client;

public sealed class IdkollenClient
{
    private readonly Transport _transport;

    internal IdkollenClient(Transport transport)
    {
        _transport = transport;
        BankIdSe = new BankIdSeEndpoint(transport);
        BankIdNo = new BankIdNoEndpoint(transport);
        Freja = new FrejaEndpoint(transport);
        MitId = new MitIdEndpoint(transport);
        Ftn = new FtnEndpoint(transport);
        Vipps = new VippsEndpoint(transport);
        Document = new DocumentEndpoint(transport);
    }

    public BankIdSeEndpoint BankIdSe { get; }
    public BankIdNoEndpoint BankIdNo { get; }
    public FrejaEndpoint Freja { get; }
    public MitIdEndpoint MitId { get; }
    public FtnEndpoint Ftn { get; }
    public VippsEndpoint Vipps { get; }
    public DocumentEndpoint Document { get; }
}
